### PR TITLE
added JsonProxy service to handle JSON formatted model queries.

### PR DESCRIPTION
**Pull request by hamersaw**
I believe this is what we talked about. Correctly formatted JSON will automatically be parsed into strongly types ModelRequest message (example below).

{
  "type" : "KMeansClustering",
  "kMeansClusteringRequest" : {
    "clusterCount" : 5
  }
}

### CODE REVIEW

**Review by drg101 — APPROVED** [review]
.proto looks perfect from the frontend!

File: src/main/java/org/sustain/server/SustainServer.java [hunk 1]
```diff
@@ -88,6 +96,65 @@ public void shutdownNow() {
...
+    static class JsonProxyService extends JsonProxyGrpc.JsonProxyImplBase {
+        @Override
+        public void modelQuery(JsonModelRequest request,
+                StreamObserver<JsonModelResponse> responseObserver) {
+            ManagedChannel channel = null;
+
+            try {
+                // open grpc channel
+                channel = ManagedChannelBuilder
+                    .forAddress("127.0.0.1", Constants.Server.PORT)
```
Comment (menuka94) [inline src/main/java/org/sustain/server/SustainServer.java:109]: Shouldn't this be populated using the ENV variable for host? @inf0rmatiker

Comment (hamersaw) [reply src/main/java/org/sustain/server/SustainServer.java]: I'm OK changing this to whatever. It seemed that parsing the config.properties file was closely tied to DB connections (being in the 'db' package. When initializing the gRPC Server with forPort(...) I'm assuming it opens on all interfaces. I have updated to use ''Util.getProperty(Constants.Server.HOST)".

File: src/main/proto/sustain.proto [hunk 2]
```diff
@@ -31,12 +43,37 @@ service Sustain{
...
+enum ModelType {
+    KMeansClustering = 0;
+    LinearRegression = 1;
+}
+
 message ModelRequest {
-  string request = 1;
+  ModelType type = 1;
+  KMeansClusteringRequest kMeansClusteringRequest = 2;
+  LinearRegressionRequest LinearRegressionRequest = 3;
```
Comment (inf0rmatiker) [inline src/main/proto/sustain.proto:54]: Can we lowercase this L to make it consistent with other var names?

File: src/main/proto/sustain.proto [hunk 3]
```diff
@@ -31,12 +43,37 @@ service Sustain{
...
 }
 
 message ModelResponse {
-  string results = 1;
+  KMeansClusteringResponse kMeansClusteringResponse = 1;
+  LinearRegressionResponse LinearRegressionResponse = 2;
```
Comment (inf0rmatiker) [inline src/main/proto/sustain.proto:59]: Same here

**Review by inf0rmatiker — CHANGES_REQUESTED** [review]
Looks awesome, very cleanly coded. See comment about changing variable names. I'll check out this branch and add the LinearRegression request model

File: src/main/java/org/sustain/server/SustainServer.java (hunk 1 above)
Comment (inf0rmatiker) [reply src/main/java/org/sustain/server/SustainServer.java]: I'll update it

**Review by inf0rmatiker — APPROVED** [review]

**Pull request merged**